website/blog/2025-06-06-powering-notification-infrastructure-novu-ferretdb.md

```diff
@@ -0,0 +1,165 @@
+By replacing MongoDB with FerretDB, you can run your notification infrastructure on a fully open-source stack without any vendor lock-in or restrictive licenses.
+This combination offers several compelling advantages:
+
+- **Open-source:** Both Novu and FerretDB are open-source, providing transparency, flexibility, and strong community support.
+- **Simplified infrastructure:** If your existing data infrastructure is already based on PostgreSQL, integrating Novu with FerretDB can simplify your database management and reduce operational overhead.
+- **Scalability and lack of vendor lock-in:** Leverage the scalability and performance characteristics of both Novu's efficient notification delivery and FerretDB without fear of vendor lock-in or licensing issues.
+
+## Connecting Novu to FerretDB
+
+Connecting Novu to your FerretDB instance is straightforward, as Novu expects a MongoDB-compatible database.
+Here's a step-by-step guide to get you started:
+
+1. **Ensure FerretDB is running:** Make sure your FerretDB instance is active and accessible.
+   If you haven't set it up yet, refer to our [FerretDB Installation Guide](https://docs.ferretdb.io/installation/ferretdb/).
+2. **Self-host Novu:** Novu offers self-hosting options, typically via Docker or by running individual services.
+   You'll need to set up Novu – [refer to Novu's documentation](https://docs.novu.co/community/self-hosting-novu/overview) for detailed instructions.
+3. **Configure Novu's database connection:** When configuring Novu, instead of pointing to a MongoDB instance, update the `.env` file to point to your FerretDB instance.
+
+   You'll need to update the `MONGO_URL` environment variable, as shown below (assuming FerretDB is running on `127.0.0.1:27017`):
+
+   ```text
+   MONGO_URL=mongodb://<username>:<password>@127.0.0.1:27017/novu-db
+   ```
+
+   Ensure to replace `<username>` and `<password>` with your FerretDB credentials.
+
+4. **Launch Novu and test:** Once configured, launch Novu.
+   It should connect to FerretDB, create the necessary collections, and be ready to start sending notifications!
+
+   You can then proceed with Novu's setup, creating workflows and sending test notifications.
+
+### Example of Novu with FerretDB as the backend
+
+Below is a screenshot of the Novu dashboard with a notification workflow created to send an in-app notification to a user:
+
+![A screenshot of a notification workflow in the Novu dashboard with FerretDB as the backend](/img/blog/novu-workflow-dashboard.png)
+
+The next image shows the in-app notification received by a user:
+
+![A screenshot of a Novu in-app notification received by a user](/img/blog/novu-notification.png)
+
+Here is a view of all the collections created in FerretDB for Novu as well as the workflow and notifications created:
+
+```text
+> use novu-db
+switched to db novu-db
+> show collections
```

Comment:
Nothing in this output suggests that it is FerretDB and not MongoDB. Let's run `buildInfo` or `serverStatus` command.